data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/ConnectionConfiguration.java

```diff
@@ -5,31 +5,33 @@
 package org.opensearch.dataprepper.plugins.source.opensearch.configuration;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+
 import java.nio.file.Path;
+import java.time.Duration;
 
 public class ConnectionConfiguration {
 
   @JsonProperty("cert")
   private Path certPath;
 
   @JsonProperty("socket_timeout")
-  private Integer socketTimeout;
+  private Duration socketTimeout;
```

Comment:
There aren't for the opensearch sink, so I didn't include any. We can add later if we find the need.

Reply:
What will we use without a default value? If we rely on a library and the default changes between major version, this could be a breaking change for us if we don't update data prepper major version. We should define one to prevent breaking changes.

Reply:
OpenSearch sink appears to use the defaults that come with the client by default. Do we not like this approach? If not how do we determine defaults?

Reply:
If they change the default in the library/client, maybe it is for a reason though?